cmd/ocsp-updater/main.go

```diff
@@ -23,20 +28,80 @@ import (
 	"github.com/letsencrypt/boulder/sa"
 )
 
-// FatalError indicates the updater should stop execution
-type FatalError string
-
-func (e FatalError) Error() string { return string(e) }
-
 // OCSPUpdater contains the useful objects for the Updater
 type OCSPUpdater struct {
 	stats statsd.Statter
 	log   *blog.AuditLogger
-	cac   rpc.CertificateAuthorityClient
+
+	rpcMu *sync.RWMutex
+
 	dbMap *gorp.DbMap
+
+	cac core.CertificateAuthority
+	pub core.Publisher
+
+	clk clock.Clock
+
+	ocspMinTimeToExpiry time.Duration
+	oldestIssuedSCT     time.Duration
+
+	newCertWindow    time.Duration
+	oldOCSPWindow    time.Duration
+	missingSCTWindow time.Duration
+	newCertBatch     int
+	oldOCSPBatch     int
+	missingSCTBatch  int
+
+	numLogs int
+}
+
+// This is somewhat gross but can be pared down a bit once the publisher and this
+// are fully smooshed together
+func newUpdater(log *blog.AuditLogger, stats statsd.Statter, clk clock.Clock,
+	dbMap *gorp.DbMap, ca core.CertificateAuthority, pub core.Publisher,
+	numLogs int, config cmd.OCSPUpdaterConfig) (*OCSPUpdater, error) {
+	updater := OCSPUpdater{
```

Comment:
This is a very large struct with many methods. Can we split these up so that the ct logging code had just what it needs and the ocspResponse generating code has what it needs?

I'm fine with there being an updater struct returned here that calls methods on the other two structs. I think you'll find that it's easier to initialize the structs, as well.